brownie/src/index.ts

```diff
@@ -0,0 +1,162 @@
+import pg, { PoolConfig } from "pg"
+
+import { logger } from "./logger.js"
+import { readParams } from "./bootstrap.js"
+import { Config } from "./config.js"
+import { ResourceStatus, evaluateResource } from "./utils.js"
+import Kafkajs from "kafkajs"
+import type { KafkaConfig as KafkaJsConfig, SASLOptions } from "kafkajs"
+
+const main = async () => {
+  const config: Config = readParams()
+  logger.info("main(), Parsed configuration: \n%s", JSON.stringify({ ...config, timestampPattern: config.timestampPattern.toString() }, null, 2))
+
+  logger.info("Cleaning old databases...")
+  const cleanedDbsCount = await cleanOldDatabases(config)
+  if (cleanedDbsCount > 0) {
+    logger.info("Dropped %s database%s", cleanedDbsCount, cleanedDbsCount > 1 ? "s" : "")
+  } else {
+    logger.info("There are no databases to clean")
+  }
+  logger.info("\n\n")
+
+  logger.info("Cleaning kafka topics...")
+  const cleanedTopicsCount = await cleanTopics(config)
+  if (cleanedTopicsCount > 0) {
+    logger.info("Deleted %s topics%s", cleanedTopicsCount, cleanedTopicsCount > 1 ? "s" : "")
+  } else {
+    logger.info("There are no topics to delete")
+  }
+}
+
+const cleanOldDatabases = async (config: Config): Promise<number> => {
+  let cleanedCount = 0
+  const pgClient = new pg.Client({
+    user: config.pg.username,
+    database: config.pg.database,
+    password: config.pg.password,
+    port: config.pg.port,
+    host: config.pg.host,
+    min: 1
+  } as PoolConfig)
+
+  await pgClient.connect()
+  try {
+    const result = await pgClient.query({
+      name: "select-databases",
+      text: "SELECT d.datname as database FROM pg_catalog.pg_database d WHERE d.datname like $1 ORDER BY d.datname",
+      values: ["%pit%"]
+    })
+    for (let row of result?.rows) {
+      const dbname = row.database
+
+      try {
+        const status = evaluateResource(dbname, config.timestampPattern, new Date(), config.retentionMinutes)
```

Comment:
- Selecting the dbs from `pg_database` and assuming that the db will have timestamp in the db name can lead to missing some dbs from clean up if the timestamp is not used in the dbname. 
- And I see this is where `evaluateResource` fn derives the timestamp from the dbname and check if  `current time  - time of db creation derived from dbname < config.retentionMinutes` 

I was thinking why not we just use this query to get the `modified date of the db` 

```sql
SELECT (pg_stat_file('base/'||oid ||'/PG_VERSION')).modification modified_date, datname FROM pg_database
WHERE d.datname like $1 
ORDER BY modified_date
```

This will give the modified date, which can be used against the current time to check if it is over the retentionMinutes?

Reply:
I have explained this aspect in a new readme. The idea was to have a simple tool the logic which can be applied to postgres, elastic, kafka and couch db and etc.

The query you are showing requires elevated privileges, so dev need to deal with devops to arrange that. We use embedded pattern for all resources and call it a day. At least in the MVP version.